subworkflows/local/minimap2_subworkflow.nf

```diff
@@ -19,8 +19,15 @@ workflow MINIMAP2_SUBWORKFLOW {
     ch_versions = ch_versions.mix(MINIMAP2_INDEX.out.versions)
     ch_index = MINIMAP2_INDEX.out.index
 
+    // match index to the corresponding reads
+    ch_reads = reads.map{meta, reads -> [meta.id, meta,reads]}
+    ch_mapping = ch_index
+        .map{meta, index -> [meta.id, meta, index]}
+        .combine(ch_reads, by:0)
+        .map{id, index_meta, index, reads_meta, reads -> [index_meta, index, reads_meta, reads]}
```

Comment:
I don' totally understand what's happening here at a high level, and why there ends up being four components

Reply:
it ends up being four components because the "meta" for the index and reads has to be separate and tied to the name of the file for each of those. The index = the assembly here which might not be clear, but I call it the index because in the previous step it gets "indexed." So this ensures that the right reads are mapping to the right assembly and not just random pairs